### PR TITLE
Split filters.py

### PR DESCRIPTION
**Pull request by JevinJ**
### Overview

Preserves history.

Split filters.py into algorithm.py, compositefilters.py, polydatafilters.py, structuredgridfilters.py, uniformgridfilters.py, unstructuredgrid.py

Moves the above into /core/filters/

TODO: Docstring changes for the separated files.

### CODE REVIEW

**Comment by akaszynski** [comment]
@JevinJ, could you wait until #1163 is merged (or abandoned)?  Last merge wasn't too bad, but since it's a big change I don't want to mess it up.

**Comment by JevinJ** [comment]
@akaszynski This is ready except for Sphinx errors which I'm not sure about and file docstrings need to be changed, you'd probably be a better author for those.

**Comment by JevinJ** [comment]
@tkoyama010 Thanks for looking at a fix, I've been too distracted to come back to this.

**Comment by tkoyama010** [comment]
@JevinJ It's my pleasure :-)

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -539,7 +539,7 @@ def download_kitchen(split=False):
         alg.SetInputDataObject(mesh)
         alg.SetExtent(extent)
         alg.Update()
-        result = pyvista.filters._get_output(alg)
+        result = pyvista.core.filters.vtkalgorithm._get_output(alg)
```
Comment (tkoyama010) [inline pyvista/examples/downloads.py:542]: @JevinJ It seems that building document error occured here. Do you have any idea?

Comment (JevinJ) [reply pyvista/examples/downloads.py]: I made a vtkalgorithm class at first, but I decided there wasn't a good enough reason for it yet. This should be fixed.

**Comment by JevinJ** [comment]
@pyvista/developers This is done except for example section for file docstrings. They're currently copied from the original file. I've never used filters so I'm not a good author for these.

**Comment by adeak** [comment]
Speaking of docstrings, I noticed a while back that some of the filter docstrings copy vtk, referencing features that aren't implemented yet. For instance in [`extrude_rotate()`](https://github.com/pyvista/pyvista/pull/1165/files#diff-ad5c36384da34894025fc614b8bdb8d233763357ed90cc24998ebc39e709da97R1832-R1840) or [`extrude()`](https://github.com/pyvista/pyvista/pull/1165/files#diff-ad5c36384da34894025fc614b8bdb8d233763357ed90cc24998ebc39e709da97R1771-R1774):
```
        There are a number of control parameters for this filter. You
        can control whether the sweep of a 2D object (i.e., polygon or
        triangle strip) is capped with the generating geometry via the
        "Capping" parameter.
```
I know this is tangential to this PR, but I haven't raised this yet elsewhere so I just wanted to note it so others are aware as well.

**Comment by akaszynski** [comment]
There are quite a few duplicates, especially in `datasetfilters.py`. For fun, run:
```
pylint --disable=all --enable=similarities
```

For example:
https://github.com/pyvista/pyvista/blob/35dd56bed170bc0fcc32529e0364fdb217a84d8e/pyvista/core/filters/datasetfilters.py#L3102

versus:

https://github.com/pyvista/pyvista/blob/35dd56bed170bc0fcc32529e0364fdb217a84d8e/pyvista/core/filters/polydatafilters.py#L272

**Comment by JevinJ** [comment]
> 
> 
> There are quite a few duplicates, especially in `datasetfilters.py`. For fun, run:
> 
> ```
> pylint --disable=all --enable=similarities
> ```
> 
> For example:
> 
> https://github.com/pyvista/pyvista/blob/35dd56bed170bc0fcc32529e0364fdb217a84d8e/pyvista/core/filters/datasetfilters.py#L3102
> 
> versus:
> 
> https://github.com/pyvista/pyvista/blob/35dd56bed170bc0fcc32529e0364fdb217a84d8e/pyvista/core/filters/polydatafilters.py#L272

Err, somehow the other classes came back to datasetfilters.py(or I just forgot to remove them which is possible). Good catch, I've removed the duplicate classes.

**Comment by akaszynski** [comment]
I'm afraid we're going to run into several issues here.  First, `master` and this branch diverge in several methods that have already been refactored, for example, `plot_curvature` and `multi_ray_trace`.  Secondly, #1229 will be unable to be merged without a significant rewrite.

I'm hesitant to say this, especially knowing how much work went into this, but for these refactors we will really need to put any feature development on hold and wait until the PR is complete.  Additionally, we should do this one filters class at a time so the PR can be created and merged in a very short period of time.  Since we've merged several PRs over the past 6 weeks, any modifications to filters will be lost with this PR, so it's best for it to be done over a very short period of time.

Finally, I feel like the module names are a bit long and should either be shortened to remove the redundant "filters" or should have some underscores added (or both).

Again, I would really prefer to merge this, but I also worry about merging other PRs and the about the loss of any work done over the past weeks.  Huge refactors like this are very dangerous and I think breaking them up would be best.

**Pull request closed without merging**